src/tests/test_main.py

```diff
@@ -17,3 +17,18 @@ def test_read_latest():
 
     r = MultipleGSP(**response.json())
     assert len(r.gsps) == 339
+
+def test_floor_30_minutes():
+    """
+    Test if floor_30_minutes_dt method works by testing against every minute in a hour
+    For minutes in range [0, 30) => Will floor to 0 minutes
+    For minutes in range [30, 60) => Will floor to 30 minutes
+    """
+    list_of_time = get_every_minute()
+    for time in list_of_time:
+        floor_minute = floor_30_minutes_dt(time)
+        if time.minute < 30:
+            assert floor_minute.minute == 0
+        else:
+            assert floor_minute.minute == 30
+
```

Comment:
I really like this, thanks so much for doing this. This should cover all times which is amazing. 

Would you be able to add one other example that shows it explicity i.e 

`
in = datetime.fromisoformat('2019-01-01 12:15:05')
out = datetime.fromisoformat('2019-01-01 12:00:00')
assert floor_30_minutes_dt(in) == out 
`

- or something like that

Reply:
Yes I can do explictly too if you want, but I will make another pull request when I have some time.
I'm also Software engineer full time so I do contribute in my spear time 😄

Reply:
No problem - I totally understand - Thank you for doing this. I'll add comment to the issue.